### PR TITLE
chore: upgrade action deps

### PR DESCRIPTION
**Pull request by chenrui333**
- upgrade actions to base on nodejs 20 runtime

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -18,12 +18,10 @@ jobs:
 
     runs-on: ${{ matrix.os }}
     steps:
-      - name: Checkout code
-        uses: actions/checkout@v2
-        with:
-          fetch-depth: 0
```
Comment (facundobatista) [inline .github/workflows/tests.yaml:24]: Hello! Thanks for this! Question, why the removal of `fetch-depth`? Thanks!

Comment (chenrui333) [reply .github/workflows/tests.yaml:24]: Removing the fetch-depth would just do the HEAD build, would it disrupt anything?

Comment (facundobatista) [reply .github/workflows/tests.yaml:24]: Ah, I see

**Review by facundobatista — APPROVED** [review]
Thanks!!

**Pull request merged**